internal/rpcapi/stacks.go

```diff
@@ -697,11 +697,19 @@ func stackChangeHooks(send func(*terraform1.StackChangeProgress) error, mainStac
 		// When Terraform core reports a resource instance plan status, we
 		// forward it to the events client.
 		ReportResourceInstanceStatus: func(ctx context.Context, span any, rihd *hooks.ResourceInstanceStatusHookData) any {
+			// addrs.Provider.String() will panic if called on the zero value.
+			// Which we shouldn't ever be holding at this point!! So this
+			// paranoia is probably unnecessary.
+			providerAddr := rihd.Addr.Item.ResourceInstance.Resource.Resource.ImpliedProvider()
```

Comment:
I'd mildly prefer an empty/null value over an implied provider value here. Using the implied provider might paper over a more significant bug, and I think it would be better to surface the missing data to the consumer of this stream to detect and fix that bug. Not a blocker, this approach is okay too.

Reply:
That's sound logic, I'll change it to do that.